packages/common/src/index.ts

```diff
@@ -1094,11 +1094,11 @@ export type Project = {
 };
 
 export type CreateProject = Omit<Project, 'projectUuid'> & {
-    warehouseConnection?: CreateWarehouseCredentials;
+    warehouseConnection: CreateWarehouseCredentials;
 };
 
 export type UpdateProject = Omit<Project, 'projectUuid'> & {
-    warehouseConnection?: CreateWarehouseCredentials;
+    warehouseConnection: CreateWarehouseCredentials;
```

Comment:
Are these needed on update? User could just update the dbt project settings? Or are we treating them all as one for now?

Reply:
We are treating them as one currently, the only reason we had this as optional was because dbt cloud and remote rpc didn't have the warehouse credentials. I know we want to improve the user experience by not requesting the password/secret values but don't want to bring that work into this PR.